DTSPO-14919 - Passthrough url exclusions (#1157)
hmcts/cnp-jenkins-library: README.md

*Performance tests use Gatling. You can find more information about the tool on their website https://gatling.io/.

You can passthrough options to the frontend security scan script by using the URL_EXCLUSIONS variable in your Jenkinsfile. This allows you to customise zap proxy scans of your application.

> If you use gradle for your frontend application, add variable called `IS_FRONTEND` to your Jenkinsfile and set its value to `true`.

Reviewer: provide an example please, this isn't very clear as-is

*Performance tests use Gatling. You can find more information about the tool on their website https://gatling.io/.

You can passthrough options to the frontend security scan script by using the URL_EXCLUSIONS variable in your Jenkinsfile. This allows you to customise zap proxy scans of your application.

> You can find an example [here](https://github.com/hmcts/idam-web-public/commit/aa5a40019959641ea1f355a9b6b66b56c99d34e0)

Reviewer: I wouldn't make it a parameter, I would just hardcode it in the Jenkinsfile_nightly, simply the easiest to understand

Reviewer: yeah

> You can find an example [here](https://github.com/hmcts/idam-web-public/commit/8ffba3b26b0485c4a63679b6263b954e09c6909c#diff-539d7df7a08b69557a3bb00c6fc85d8dda08d43310f75214ddb2019124d41cfdR62)

Reviewer: ```suggestion
You can find an example in [idam-web-public](https://github.com/hmcts/idam-web-public/commit/8ffba3b26b0485c4a63679b6263b954e09c6909c#diff-539d7df7a08b69557a3bb00c6fc85d8dda08d43310f75214ddb2019124d41cfdR62)
```

If you use gradle for your frontend application, add variable called `IS_FRONTEND` to your Jenkinsfile and set its value to `true`.

> You can find an example [here](https://github.com/hmcts/idam-web-public/commit/f06fe5b0a3dd34326374c05ad0b90871ed0bb678#diff-539d7df7a08b69557a3bb00c6fc85d8dda08d43310f75214ddb2019124d41cfdR64)

Reviewer: ```suggestion
You can find an example in [idam-web-public](https://github.com/hmcts/idam-web-public/commit/f06fe5b0a3dd34326374c05ad0b90871ed0bb678#diff-539d7df7a08b69557a3bb00c6fc85d8dda08d43310f75214ddb2019124d41cfdR64)
```

*Performance tests use Gatling. You can find more information about the tool on their website https://gatling.io/.

> You can passthrough options to the frontend security scan script by using the URL_EXCLUSIONS variable in your Jenkinsfile. This allows you to customse zaproxy scans of your application.

Reviewer: ```suggestion
You can passthrough options to the frontend security scan script by using the URL_EXCLUSIONS variable in your Jenkinsfile. This allows you to customise zap proxy scans of your application.
```

*Performance tests use Gatling. You can find more information about the tool on their website https://gatling.io/.

> You can passthrough options to the frontend security scan script by using the ZAP_URL_EXCLUSIONS variable in your Jenkinsfile. This allows you to customise zap proxy scans of your application.

Reviewer 2: Docs need updating with new approach ?

Author: On hmcts.github.io you mean?

Reviewer 2: I just meant the README. It says "You can passthrough options to the frontend security scan script by using the ZAP_URL_EXCLUSIONS variable in your Jenkinsfile" . shouldn't be an optional parameter to `enableSecurityScan` instead?

Author: Updated
